### PR TITLE
*: Support the statement of "create table ... like"

### PR DESCRIPTION
**Pull request by zimulala**
Fix issue #2668

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -571,6 +571,45 @@ func (d *ddl) buildTableInfo(tableName model.CIStr, cols []*table.Column, constr
 	return
 }
 
+func (d *ddl) CreateTableWithLike(ctx context.Context, ident, referIdent ast.Ident) error {
+	is := d.GetInformationSchema()
+	_, ok := is.SchemaByName(referIdent.Schema)
+	if !ok {
+		return infoschema.ErrTableNotExists.GenByArgs(referIdent.Schema, referIdent.Name)
```
Comment (hanfei1991) [inline ddl/ddl_api.go]: DBNotExists

Comment (zimulala) [inline ddl/ddl_api.go]: mysql> create table x like testttt.t;
ERROR 1146 (42S02): Table 'testttt.t' doesn't exist
mysql> create table x like test.ttttt;
ERROR 1146 (42S02): Table 'test.ttttt' doesn't exist

File: ddl/ddl_db_test.go [hunk 2]
```diff
@@ -894,6 +894,50 @@ func (s *testDBSuite) TestUpdateMultipleTable(c *C) {
...
+	s.tk.MustExec("insert into t set c2=1")
+	s.tk.MustExec("create table t1 like test.t")
+	s.tk.MustExec("insert into t1 set c2=11")
+	s.tk.MustQuery("select * from t").Check(testkit.Rows("10 1"))
+	s.tk.MustQuery("select * from t1").Check(testkit.Rows("1 11"))
+	ctx := s.tk.Se.(context.Context)
+	is := sessionctx.GetDomain(ctx).InfoSchema()
+	tbl, err := is.TableByName(model.NewCIStr("test"), model.NewCIStr("t1"))
+	c.Assert(err, IsNil)
+	c.Assert(tbl.Meta().ForeignKeys, IsNil)
```
Comment (hanfei1991) [inline ddl/ddl_db_test.go]: How about the index ?

**Comment by coocood** [comment]
LGTM

**Pull request merged**